pkg/nsx/services/securitypolicy/builder.go

```diff
@@ -800,22 +803,41 @@ func (service *SecurityPolicyService) buildRuleBasicInfo(obj *v1alpha1.SecurityP
 	if err != nil {
 		return nil, err
 	}
+	display_name := fmt.Sprintf("%s-%d-%d", service.buildRuleName(obj, rule, ruleIdx), portIdx, portAddressIdx)
+	if createdFor == common.ResourceTypeNetworkPolicy {
+		// The internal security policy rule converted from network policy has its own name generated by the ingress/egress ports.
+		// e.g. input: internal security policy's rule name: TCP.http-UDP.1234-ingress-allow
```

Comment:
Let's assume the input 
      ports:
        - protocol: TCP
          port: 80
        - protocol: UDP
          port: 1234
          endPort: 1235

From the function: buildRulePortsString, to my understanding, the generated rule name is:
TCP.80-UDP.1234.1235-ingress-allow for input internal security policy rule,
and then it will be converted to two rules:
TCP.80-UDP.1234.1235.TCP.80-ingress-allow and TCP.80-UDP.1234.1235.UDP.1234-ingress-allow
is it correct?